database/user_test.go

```diff
@@ -0,0 +1,42 @@
+package database
+
+import (
+	"testing"
+	"time"
+)
+
+//TestMonthStart ensures we calculate the start of the subscription month
+// correctly.
+func TestMonthStart(t *testing.T) {
+	now := time.Now()
+	tests := []struct {
+		subUntil     time.Time
+		startOfMonth time.Time
+	}{
+		{
+			// If the monthly sub cycle reset yesterday, expect the start of
+			// month to also be yesterday.
+			subUntil:     time.Date(2020, 1, now.Day()-1, 3, 4, 5, 6, time.UTC),
+			startOfMonth: time.Date(now.Year(), now.Month(), now.Day()-1, 0, 0, 0, 0, time.UTC),
+		},
+		{
+			// If the monthly sub cycle resets today, expect the start of
+			// month to also be today.
+			subUntil:     time.Date(2020, 1, now.Day(), 3, 4, 5, 6, time.UTC),
+			startOfMonth: time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, time.UTC),
+		},
+		{
+			// If the monthly sub cycle resets tomorrow, expect the start of
+			// month to be tomorrows date but a month back.
+			subUntil:     time.Date(2020, 1, now.Day()+1, 3, 4, 5, 6, time.UTC),
+			startOfMonth: time.Date(now.Year(), now.Month()-1, now.Day()+1, 0, 0, 0, 0, time.UTC),
+		},
+	}
```

Comment:
why are you hardcoding 2020 in the `subUntil`s?

Reply:
The idea was to have something in the past, a subscription that's been running for a while. I can also use `now.Year()-1` if you prefer.